File: app/src/main/java/ru/systempla/weatherapp/di/AppComponent.java
package ru.systempla.weatherapp.di;

import javax.inject.Singleton;

import dagger.Component;

import ru.systempla.weatherapp.di.module.AppModule;
import ru.systempla.weatherapp.di.module.CiceroneModule;
import ru.systempla.weatherapp.di.module.LocationGetterModule;
import ru.systempla.weatherapp.di.module.RepoModule;
import ru.systempla.weatherapp.di.module.SettingsModule;
import ru.systempla.weatherapp.mvp.presenter.MainPresenter;
import ru.systempla.weatherapp.mvp.presenter.WeatherDataPresenter;
import ru.systempla.weatherapp.mvp.view.ui.MainActivity;
import ru.systempla.weatherapp.mvp.view.ui.fragment.ForecastFragment;
import ru.systempla.weatherapp.mvp.view.ui.fragment.WeatherDataFragment;

@Singleton
@Component(modules = {
        RepoModule.class,
        AppModule.class,
        LocationGetterModule.class,
        SettingsModule.class,
        CiceroneModule.class
})

public interface AppComponent {
    void inject(MainPresenter presenter);
    void inject(MainActivity mainActivity);
    void inject(WeatherDataPresenter weatherDataPresenter);
    void inject(WeatherDataFragment weatherDataFragment);
    void inject(ForecastFragment forcastFragment);
}

File: app/src/main/java/ru/systempla/weatherapp/di/AppComponent.kt
package ru.systempla.weatherapp.di

import com.tbruyelle.rxpermissions3.RxPermissions
import dagger.Component
import ru.systempla.weatherapp.di.module.*
import ru.systempla.weatherapp.mvp.presenter.ForecastPresenter
import ru.systempla.weatherapp.mvp.presenter.MainPresenter
import ru.systempla.weatherapp.mvp.presenter.WeatherDataPresenter
import ru.systempla.weatherapp.mvp.view.ui.MainActivity
import ru.systempla.weatherapp.mvp.view.ui.fragment.ForecastFragment
import ru.systempla.weatherapp.mvp.view.ui.fragment.WeatherDataFragment
import javax.inject.Singleton

@Singleton
@Component(modules =
    [RepoModule::class,
    LocationGetterModule::class,
    SettingsModule::class,
    CiceroneModule::class]
)
interface AppComponent {
    fun inject(presenter: MainPresenter)
    fun inject(mainActivity: MainActivity)
    fun inject(weatherDataPresenter: WeatherDataPresenter)
    fun inject(weatherDataFragment: WeatherDataFragment)
    fun inject(forecastPresenter: ForecastPresenter)
    fun inject(forecastFragment: ForecastFragment)
}